VERSIONING.rst

```diff
@@ -2,18 +2,30 @@
 Versioning Policy
 *****************
 
-We will use a three-part X.Y.Z (Major.Minor.Patch) versioning definition with the following meanings.
+We use a three-part X.Y.Z (Major.Minor.Patch) versioning definition as follows:
 
-* X (Major) version changes cover changes to the code-base that are expected to break backwards compatibility.
-* Y (Minor) version changes cover moderate changes. These include significant (non-breaking) feature additions and might contain changes which break backwards compatability. If there are breaking changes, they will be explicitly stated in the release notes.
-* Z (Patch) version changes cover small changes. They will not break backwards compatibility.
+* **X (Major)** version changes are significant and expected to break backwards compatibility.
+* **Y (Minor)** version changes are moderate changes. These include:
+
+  * Significant non-breaking feature additions.
+  * Potentially backwards incompatible changes. Any such changes will be explicitly stated in the release notes.
+  * Changes to our package's declared dependency versions.

```

Comment:
This is the old language, there's a few changes since. The newest version is:

>Any change to the version of a dependency.

Is that clear enough or would you still prefer changes?

Reply:
Do we do a minor rev on patch changes to dependencies?

Reply:
Under this contract, yes.

That seems sensible to me for the following reasons:

1. Versioning policies differ from dependency to dependency, so we can't make reliable predictions on how our dependencies will change, only on how we'll notify our customers of changes.
1. Dependency updates tend to happen in clusters, so I expect to see things like 2-3 dependencies having patch rev updates in one of our changes.
1. Part of the goal here is to be clear, predictable, and as safe as possible. Changing dependencies always carries some risk. Doing a minor version rev every time adds some noise, but provides a reliable, clear bound and helps ensure Z updates are always safe.

Reply:
Following on from this morning's discussion, if we want to start being more prescriptive about our stated support for max versions as well, then I think what we're wanting to convey here is that we will minor version bump on two events:
* we change the *required minimum* version of a dependency
* we introduce a new dependency

Though in writing that up, I remembered that Maven only supports explicit version statements, not ranges...

Reply:
>if we want to start being more prescriptive about our stated support for max versions as well

If we do, then why wouldn't we change the minor version if we change our max supported version of a dependency? That's a change to what we support that can cause integration work for customers, if we need to (for example) pin to an earlier version of a library pending a fix.

All of these cases seem to me like they still fall cleanly under "Any change to the version of a dependency.", which is also language/platform/packaging system-agnostic.

Reply:
What I worry about there is the different release cultures that different languages have and how that will affect the versioning we advertise. If we assume the dependency releases are rare, then sure, any changes to advertised support for dependency versions should be fine. But if dependency releases are frequent and we want to be explicit about what dependency versions we support, I worry that bumping the minor version every time that advertised support set changes will result in our minor version changes becoming frequent events, and in turn making it more difficult for consumers of our libraries to keep up to date.

I guess this also stems from the very different ways that dependency versions are handled in different languages and distribution systems. For cases like Maven where we explicitly state the version we expect, the relationship is necessarily much different than cases like common PyPI usage where we instead state a closure of versions that we expect to work.

Reply:
No more concerns from me!